src/planners/path_search.jl

```diff
@@ -56,14 +56,53 @@ function Base.copy(sol::PathSearchSolution)
                               search_tree, search_frontier, search_order)
 end
 
-Base.iterate(sol::PathSearchSolution) = iterate(sol.plan)
-Base.iterate(sol::PathSearchSolution, istate) = iterate(sol.plan, istate)
-Base.getindex(sol::PathSearchSolution, i::Int) = getindex(sol.plan, i)
-Base.length(sol::PathSearchSolution) = length(sol.plan)
+"""
+Solution type for search-based planners that produce fully ordered plans.
+"""
+mutable struct BiPathSearchSolution{S<:State,T} <: OrderedSolution
+    status::Symbol
+    plan::Vector{Term}
+    trajectory::Union{Vector{S},Nothing}
+    expanded::Int
+    f_search_tree::Union{Dict{UInt,PathNode{S}},Nothing}
+    f_frontier::T
+    f_expanded::Int
+    f_trajectory::Union{Vector{S},Nothing}
+    b_search_tree::Union{Dict{UInt,PathNode{S}},Nothing}
+    b_frontier::T
+    b_expanded::Int
+    b_trajectory::Union{Vector{S},Nothing}
+end
+
+BiPathSearchSolution(status::Symbol, plan) =
+    BiPathSearchSolution(status, plan, nothing, -1, nothing, nothing, -1, nothing, nothing, nothing, -1, nothing)
+BiPathSearchSolution(status::Symbol, plan, trajectory) =
+    BiPathSearchSolution(status, plan, trajectory, -1, nothing, nothing, -1, nothing, nothing, nothing, -1, nothing)
+
+function Base.copy(sol::BiPathSearchSolution)
+    fields = map(fieldnames(BiPathSearchSolution)) do field 
+        x = getfield(sol, field)
+        (x isa Symbol || isnothing(x)) && return(x)
+        copy(x)
+    end
+    return BiPathSearchSolution(fields...)
+end
+
+const SearchSolutions = Union{PathSearchSolution, BiPathSearchSolution}
+
+function Base.show(io::IO, sol::SearchSolutions)
+    pl = sol.status == :success ? length(sol.plan) : "-"
+    println(io, "BiPathSearchSolution: (", sol.status, ", ", pl, ", ", sol.expanded, ")")
```

Comment:
This is nice to have! But can we:

- Change it to display the name of the type, rather than hardcoding `"BiPathSearchSolution"`? I think you can just do this with `typeof(sol)`.

- Display the plan even if the status is not `:success`? There are cases where the partial plan is helpful, even if you reach the `:max_time` or `:max_nodes` termination criterion. The only case where we shouldn't display the plan is when `sol.status == :failure`.

Reply:
Oh, also, it'd be good to display not just the length of the plan but the plan itself, and maybe indicate clearly what the numbers etc refer to.

Perhaps let's just remove this for now, I want to think a bit more about the best format for displaying a summary of the information.

Reply:
Removed.